runners/spark/src/main/java/org/apache/beam/runners/spark/io/SourceDStream.java

```diff
@@ -60,47 +59,64 @@
   private final UnboundedSource<T, CheckpointMarkT> unboundedSource;
   private final SparkRuntimeContext runtimeContext;
   private final Duration boundReadDuration;
+  private final int numPartitions;
```

Comment:
Since this is a member of the DStream it is initialized once, and would be recovered on checkpoint recovery, right ?
So it should be noted here that this is a "one time" set of partitions/splits, and cannot change throughout the entire life of the application (+ recovery/resume).